### PR TITLE
refactor: move cli-only js to cli

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: cli/build.rs [hunk 1]
```diff
@@ -1,306 +1,410 @@
...
+  use super::*;
+  use crate::deno_webgpu_get_declaration;
+  use deno_core::error::{custom_error, AnyError};
+  use deno_core::{op, Extension, JsRuntime, OpState, RuntimeOptions};
+  use regex::Regex;
+  use serde::Deserialize;
+  use serde_json::{json, Value};
+  use std::collections::HashMap;
+
+  // TODO(bartlomieju): this module contains a lot of duplicated
```
Comment (bartlomieju) [inline cli/build.rs:19]: Suggest to split this file into `build/mod.rs`, `build/tsc.rs` and `build/cli.rs`

File: cli/worker.rs [hunk 2]
```diff
@@ -454,16 +525,36 @@ pub async fn create_main_worker(
...
+    js_run_benchmarks_callback,
+    js_enable_test_callback,
+    js_enable_bench_callback,
+  ) = {
+    let scope = &mut worker.js_runtime.handle_scope();
+    (
+      grab_cb(scope, "__bootstrap.testing.runTests"),
+      grab_cb(scope, "__bootstrap.testing.runBenchmarks"),
+      grab_cb(scope, "__bootstrap.testing.enableTest"),
+      grab_cb(scope, "__bootstrap.testing.enableBench"),
```
[the remark this answers is not in the document]
Comment (nayeemrmn) [reply cli/worker.rs]: It's strange to compromise that cli wouldn't have access to the transient bootstrapping namespace for this purpose, especially if the very bootstrapping code is being moved to `cli/js` here...

Looks like we can expand the `Worker::bootstrap_from_options()` call (2 lines, no private API) and put the grab_cb stuff in between.

File: cli/build.rs [hunk 3]
```diff
@@ -440,11 +544,13 @@ fn main() {
...
 
-  // Main snapshot
   let compiler_snapshot_path = o.join("COMPILER_SNAPSHOT.bin");
-
   let js_files = get_js_files("tsc");
-  create_compiler_snapshot(&compiler_snapshot_path, js_files, &c);
+  ts::create_compiler_snapshot(&compiler_snapshot_path, js_files, &c);
+
+  let cli_snapshot_path = o.join("CLI_SNAPSHOT.bin");
+  let js_files = get_js_files("js");
```
Comment (nayeemrmn) [inline cli/build.rs:552]: I would expect the new `cli/js` files to be included through a simple extension added to [`cli_exts()`](https://github.com/denoland/deno/blob/89c5aa85984eabd300818f8275e3970f12a89754/cli/ops/mod.rs#L14), would that avoid a lot of these changes?

**Comment by nayeemrmn** [comment]
`MainWorker::bootstrap()` in `runtime/worker.rs` calls behaviour now defined in `cli/js`, that should be addressed

**Comment by crowlKats** [comment]
closing in favour of #16597

**Pull request closed without merging**